compiler/rustc_ast_lowering/src/lib.rs

```diff
@@ -717,7 +718,7 @@ impl<'a, 'hir> LoweringContext<'a, 'hir> {
                 v.insert(local_id);
                 self.item_local_id_counter.increment_by(1);
 
-                assert_ne!(local_id, hir::ItemLocalId::new(0));
+                debug_assert_ne!(local_id, hir::ItemLocalId::new(0));
```

Comment:
Hot (called for most nodes in code), but the assert is trivial.
Not sure about usefulness of this assert.